x/poolmanager/taker_fee.go

```diff
@@ -148,8 +148,7 @@ func (k Keeper) chargeTakerFee(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDenom
 // returns (1 - takerFee) * tokenIn, takerFee * tokenIn
 func CalcTakerFeeExactIn(tokenIn sdk.Coin, takerFee osmomath.Dec) (sdk.Coin, sdk.Coin) {
 	takerFeeFactor := osmomath.OneDec().SubMut(takerFee)
-	// TODO: Remove .ToLegacyDec and instead do MulInt. Need to test state compat.
-	amountInAfterSubTakerFee := tokenIn.Amount.ToLegacyDec().MulTruncate(takerFeeFactor)
+	amountInAfterSubTakerFee := takerFeeFactor.MulIntMut(tokenIn.Amount)
```

Comment:
For my learning, why do we do a Mut here where we are just assigning the result to a new var anyway?

Reply:
it mutates takerFeeFactor in place. The assigning to a new var is just for naming convenience. If we were in rust, the original variable would no longer be usable.